src/main/scala/is/hail/expr/Type.scala

```diff
@@ -1518,66 +1521,51 @@ case class TLocus(gr: GRBase, override val required: Boolean = false) extends Co
 }
 
 object TInterval {
-  def representation(required: Boolean = false): TStruct = {
+  def representation(pointType: Type, required: Boolean = false): TStruct = {
```

Comment:
I think originally this got pulled out for all the types where the case class parameters didn't affect the representation (basically, all the GemomeReference types). Now that the representation depends on pointType, does it make sense to fold back in to the case class?